system/osx_defaults.py

```diff
@@ -160,6 +172,15 @@ def _convert_type(self, type, value):
             if not isinstance(value, list):
                 raise OSXDefaultsException("Invalid value. Expected value to be an array")
             return value
+        elif type in ["dict", "dictionary"]:
+            if isinstance(value, dict):
+                return [value]
+            elif isinstance(value, list):
+                return value
+            elif isinstance(value, str):
```

Comment:
This may be related to `type` in the `argument_spec` for an argument.  If no `type` is specified, `str` is now assumed.  Make sure your use `type="dict"`.